lite2/client.go

```diff
@@ -442,3 +613,62 @@ func (c *Client) compareNewHeaderWithRandomAlternative(h *types.SignedHeader) er
 
 	return nil
 }
+
+func (c *Client) removeNoLongerTrustedHeadersRoutine() {
+	ticker := time.NewTicker(c.removeNoLongerTrustedHeadersPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			c.RemoveNoLongerTrustedHeaders(time.Now())
+		case <-c.quit:
+			return
+		}
+	}
+}
+
+// RemoveNoLongerTrustedHeaders removes no longer trusted headers (due to
+// expiration).
+//
+// Exposed for testing.
+func (c *Client) RemoveNoLongerTrustedHeaders(now time.Time) {
+	// 1) Get the oldest height.
+	oldestHeight, err := c.trustedStore.FirstSignedHeaderHeight()
+	if err != nil {
+		c.logger.Error("can't get first trusted height", "err", err)
+		return
+	}
+
+	// 2) Get the latest height.
+	latestHeight, err := c.LastTrustedHeight()
+	if err != nil {
+		c.logger.Error("can't get last trusted height", "err", err)
+		return
+	}
+
+	// 3) Remove all headers (except the very last one) that are outside of
```

Comment:
So here we remove all expired headers in the store except the last one. This is also the one referred to by the client's `trustedHeader *types.SignedHeader`.
It is possible that there is a single expired header in the store, same as the client's `trustedHeader`. I believe any header verification from  this point on will fail as we do check for this in `Verify()`. 
But why do we need to keep it and not clean this up as well since we know it is not trusted anymore? If the client restarts with this store, it will restore this header only to be overwritten if `c.trustedHeader.Height != trustOptions.Height`. Also, shouldn't a client fail right away if initialized with an already expired header?

Reply:
> Also, shouldn't a client fail right away if initialized with an already expired header?

Yes.

> But why do we need to keep it and not clean this up as well since we know it is not trusted anymore?

because c.trustedHeader is used in VerifyHeader (both in bisection and linear verification). Removing it (setting c.trustedHeader to `nil`) can crash the light client process. We can remove the header from the store however. It should not affect the light client work.